src/nw_content.cc

```diff
@@ -597,11 +597,26 @@ void AmendManifestStringList(base::DictionaryValue* manifest,
     manifest->Set(path, pattern_list);
 }
 
+void AmendManifestList(base::DictionaryValue* manifest,
```

Comment:
This function is wrong: it doesn't merge the value in the manifest and the one you passed. See other functions start with "Amend" for example.